chronos/asyncfutures2.nim

```diff
@@ -243,6 +243,12 @@ template complete*[T](future: Future[T], val: T) =
   ## Completes ``future`` with value ``val``.
   complete(future, val, getSrcLocation())
 
+template complete*[T](F: type Future[T], val: T, fromProc: static[string] = ""): F =
+  ## Completes a new ``Future[T]`` with value ``val``.
+  let future = newFuture[T](fromProc)
```

Comment:
I think `let` should be replaced by `var`, because `complete()` operation actually changes `Future[T]` state.

Reply:
that's not correct though, `Future` is a ref object - `var` would refer to mutability in terms of switching the future variable to a different instance, not changing the contents of the future currently referred.

Reply:
```
let a = newFuture[T](fromProc)
a = newFuture[T](fromProc) # invalid
a.field = x # valid

var b = newFuture[T](fromProc)
b = newFuture[T](fromProc) # valid
b.field = x # also valid
```

the rules for `var` are a bit unfortunate for `ref object` - it's another reason to avoid `ref object` in general - this makes a lot more sense when the code spells out `ref`..